tests/generative/test_query_model.py

```diff
@@ -198,6 +198,23 @@ def run_error_test(query_engines, data, variable):
         sqlalchemy.exc.OperationalError,
         re.compile(".+Internal error: An expression services limit has been reached.+"),
     ),
+    # ARITHMETIC OVERFLOW ERRORS
+    # mssql raises this error if an operation results in an integer bigger than the max INT value
+    # or a float outside of the max range
+    # https://learn.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver16
+    # https://learn.microsoft.com/en-us/sql/t-sql/data-types/float-and-real-transact-sql?view=sql-server-ver16#remarks
+    # https://github.com/opensafely-core/databuilder/issues/1034
+    (
+        sqlalchemy.exc.OperationalError,
+        re.compile(
+            ".+Arithmetic overflow error converting expression to data type [int|float].+"
+        ),
+    ),  # arithmetic operations that result in an out-of-range int or floar
+    (
+        sqlalchemy.exc.OperationalError,
+        re.compile(".+Arithmetic overflow error for type int.+"),
```

Comment:
Do we need to wrap each error string in `re.compile`?

Reply:
yes; or at least, we need to either wrap all of them or none.  The error ignoring code iterates through these and does a `.match()` on each one, so they all need to be compiled regexes otherwise that will error.